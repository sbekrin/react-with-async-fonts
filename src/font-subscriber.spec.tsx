import * as React from 'react';
import { shallow } from 'enzyme';
import { FontSubscriber } from './index';

describe('<FontSubscriber />', () => {
  it('only accepts function child', () => {
    // Make console.error to throw to catch it with .toThrow
    global.console.error = jest.fn(reason => {
      throw new Error(reason);
    });
    expect(() => (
      <FontSubscriber>Hello</FontSubscriber>
    )).toThrowErrorMatchingSnapshot();
    // Reset console hack
    global.console.error.mockReset();
  });

  it('calls function child with fonts context', () => {
    const funcChild = jest.fn();
    shallow(<FontSubscriber>{funcChild}</FontSubscriber>, {
      context: { __fonts: { openSans: { family: 'Open Sans' } } },
    });
    expect(funcChild).toHaveBeenCalledWith({
      openSans: { family: 'Open Sans' },
    });
  });
});
