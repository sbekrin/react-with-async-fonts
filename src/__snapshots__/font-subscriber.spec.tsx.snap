// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FontSubscriber /> only accepts function child 1`] = `
"Warning: Failed prop type: Invalid prop \`children\` of type \`string\` supplied to \`FontSubscriber\`, expected \`function\`.
    in FontSubscriber"
`;
